### ISSUE TITLE
Recursive Multi-layer Perceptron

### ISSUE THREAD

**Issue by smjure** [matousc89/padasip#10]
Great module, thank you for awesome job! I'm interested in recursive regression. I wonder if Multi-layer Perceptron can be applied recursively or will get extended to recursive application at some point? As I see, at the moment there's only autoregressive option `padasip.ann.mlp.NetworkMLP.update(d)` and nothing like `padasip.filters.rls.FilterRLS.adapt(d,x)`.

**Comment by matousc89** [comment]
Thanks. I plan to remove the MLP from the Padasip with next bigger realese. There is way to many libraries providing neural networks on much better level than I can ever be able to do. So the feature you want is not available and probably never will be, because it seems to be out of scope of "simple" adaptive signal processing.

If you have different opinion, please share, maybe I am wrong. I would like to hear opinion of somebody else.

**Comment by smjure** [comment]
Am not sure if I'm eligible to give an opinion on such advanced topic. 

As an application engineer, I find your module very fast/efficient, intuitive and easy to implement. The reason I asked the question is as I am dealing with combustion, which is pretty nonlinear problem and stochastic in nature and model needs adaptation. Hence, we do not look for accuracy in the first place, but prefer fast and robust algorithm to implement adaptive control solutions. Eventually in such type of engineering problems one looks for trade-off between speed, robustness and accuracy. There are many modules out there, but I found them pretty slow (specifically ANN), hence useless for real-time applications. Maybe I did not use right keywords - in this case any recommendations to your module for adaptive nonlinear modeling very much appreciated. In general, I would avoid ANNs and similar, as there's no need for such a high complexity and where hidden extrapolation is a big problem, hence they fail in terms of robustness. But don't know, technologies are advancing fast so maybe I'm wrong and that's why is worth to try and explore. 

I hope this little comment was somewhat helpful and gave some insight from an applied engineering perspective. Thank you ones more for the great module.
